### PR TITLE
Add doxygen to the cmake build and add doxygen build to CI.

### LINKED ISSUE #289: Shall I add doxygen to the build system?
Shall I add doxygen to the build system, which would allow programmers to start using doxygen in the code?

### PR DESCRIPTION
**Pull request by edwardhartnett**
## Description

Add doxygen to the cmake build and add doxygen build to CI.

### Issue(s) addressed

Fixes #289

## Testing

No code changes. Documentation build can be checked in CI.

## Dependencies

N/A

### CODE REVIEW

**Comment by edwardhartnett** [comment]
@junwang-noaa if this can be reviewed and merged it would help with the documentation work...

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,17 @@
+# This is the main CMake file for fv3atm.
+#
+# Mark Potts, Kyle Gerheiser, Alex Richert
```
Comment (junwang-noaa) [inline CMakeLists.txt]: Is this author list correct? I think Dusan is the author.

Comment (edwardhartnett) [reply CMakeLists.txt]: Fixed.

**Review by junwang-noaa — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,3 +1,17 @@
+# This is the main CMake file for fv3atm.
+#
+# Dusan Jovic, Alex Richert
+cmake_minimum_required(VERSION 3.19)
```
Comment (junwang-noaa) [inline CMakeLists.txt:4]: A quick question, with line 4, do we still need line 17?

Comment (edwardhartnett) [reply CMakeLists.txt:4]: Good catch. I have removed line 17.

**Comment by junwang-noaa** [comment]
I tested this branch in ufs-weather-model on hera, full RT passed. 
/scratch1/NCEPDEV/nems/Jun.Wang/nems/vlab/20240118/ufs-weather-model/tests/logs/RegressionTests_hera.log

**Pull request merged**